apps/sqlserver.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apps
+
+import (
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+)
+
+type MetricsReceiverSqlServer struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+
+	confgenerator.MetricsReceiverShared `yaml:",inline"`
+}
+
+func (MetricsReceiverSqlServer) Type() string {
+	return "sqlserver"
+}
+
+func (r MetricsReceiverSqlServer) Pipelines() []otel.Pipeline {
+	cfg := map[string]interface{}{
+		"collection_interval": r.CollectionIntervalString(),
+	}
+
+	return []otel.Pipeline{{
+		Receiver: otel.Component{
+			Type:   "sqlserver",
+			Config: cfg,
+		},
```

Comment:
Do you need this local? Can you just inline the `map` below?

```suggestion
	return []otel.Pipeline{{
		Receiver: otel.Component{
			Type:   "sqlserver",
			Config: map[string]interface{}{
				"collection_interval": r.CollectionIntervalString(),
			},
		},
```

Reply:
changed